### DISCUSSION TITLE
Map roles/groups from AzureAD into a Zitadel user

### DISCUSSION THREAD

**Discussion by danielmichalichyn** [zitadel/zitadel#5153] [Q&A] [answered]
Hello,

Maybe i'm lacking knowledge in this, but any help would be appreciated. 
First of all i'll try to explain a little about our actual solution and where Zitadel can fit our needs. We have a SaaS Platform that focus primarily on factories. Right now each of our customers has several factories, and for each factory an organization is created, adding a layer of isolation between the manufacturing plants. We can give access to each factory independently per user. Customers will access our platform via SSO, and Zitadel is the tool we are trying to add for this architecture, but I have some doubts regarding the use of IDPs.

For example, a company user has access to organization A and organization B. In Zitadel's modeling, I have an organization called "CustomerY", and two other organizations called "CustomerY - Organization A" and "CustomerY - Organization B". The organization named "CustomerY" has the IDP configuration for the customer Azure AD. In AD, the user has a group called "Organization A". Can I map this AD group to an organization or role within Zitadel when logging in or creating a user?

Thanks in advance for any help!

**Comment by fforootd** [comment, marked as answer]
Hi There

Yes you can use our actions concept to update such things on the user during the login for example.

With this action you can assign a user a role in a project https://github.com/zitadel/actions/blob/main/examples/add_user_grant.js

@livio-a @adlerhurst do we provide the Idp response into the actions context? I have in my head that one our customers uses the jwtidp to do more or less the same.

**Reply by danielmichalichyn** [reply]
Yes, i'm already using actions to set a user email to verified and assign default roles. But it's seems that we don't have an object exposing the IDP response with claims and groups, or i'm not seeing it?

Reading the External Post Authentication Flow, none of the objects seems to have this information.
This information should be in the "claims" object?

![image](https://user-images.githubusercontent.com/52325588/217645259-f2773740-e634-4e2e-b181-eb0510f37c3c.png)

**Reply by fforootd** [reply]
Let me verify with @livio-a we used something similar with the jwtidp
